src/workerd/io/trace.c++

```diff
@@ -5,15 +5,159 @@
 #include <workerd/io/trace.h>
 #include <workerd/util/thread-scopes.h>
 
+#include <openssl/rand.h>
+
 #include <capnp/message.h>
 #include <capnp/schema.h>
+#include <kj/compat/http.h>
 #include <kj/debug.h>
 #include <kj/time.h>
 
 #include <cstdlib>
 
 namespace workerd {
 
+namespace tracing {
+namespace {
+kj::Maybe<kj::uint> tryFromHexDigit(char c) {
+  if ('0' <= c && c <= '9') {
+    return c - '0';
+  } else if ('a' <= c && c <= 'f') {
+    return c - ('a' - 10);
+  } else if ('A' <= c && c <= 'F') {
+    return c - ('A' - 10);
+  } else {
+    return kj::none;
+  }
+}
+
+kj::Maybe<uint64_t> hexToUint64(kj::ArrayPtr<const char> s) {
+  KJ_ASSERT(s.size() <= 16);
+  uint64_t value = 0;
+  for (auto ch: s) {
+    KJ_IF_SOME(d, tryFromHexDigit(ch)) {
+      value = (value << 4) + d;
+    } else {
+      return kj::none;
+    }
+  }
+  return value;
+}
+
+void addHex(kj::Vector<char>& out, uint64_t v) {
+  const char HEX_DIGITS[] = "0123456789abcdef";
+  for (int i = 0; i < 16; ++i) {
+    out.add(HEX_DIGITS[v >> (64 - 4)]);
+    v = v << 4;
+  }
+};
+
+void addBigEndianBytes(kj::Vector<byte>& out, uint64_t v) {
+  for (int i = 0; i < 8; ++i) {
+    out.add(v >> (64 - 8));
+    v = v << 8;
+  }
+};
+}  // namespace
+
+// Reference: https://github.com/jaegertracing/jaeger/blob/e46f8737/model/ids.go#L58
+kj::Maybe<TraceId> TraceId::fromGoString(kj::ArrayPtr<const char> s) {
+  auto n = s.size();
+  if (n > 32) {
+    return kj::none;
+  } else if (n <= 16) {
+    KJ_IF_SOME(low, hexToUint64(s)) {
+      return TraceId(low, 0);
+    }
+  } else {
+    KJ_IF_SOME(high, hexToUint64(s.slice(0, n - 16))) {
+      KJ_IF_SOME(low, hexToUint64(s.slice(n - 16, n))) {
+        return TraceId(low, high);
+      }
+    }
+  }
+  return kj::none;
+}
+
+// Reference: https://github.com/jaegertracing/jaeger/blob/e46f8737/model/ids.go#L50
+kj::String TraceId::toGoString() const {
+  if (high == 0) {
+    kj::Vector<char> s(17);
+    addHex(s, low);
+    s.add('\0');
+    return kj::String(s.releaseAsArray());
+  } else {
```

Comment:
Nit: This pattern is repeated almost all of workerd, and I would like to raise my voice even though you're just moving the code. 

----

This else is not needed since previous if statement already returns. 

----

I can finally rest.